### PR TITLE
Added bash completion for --filter & --format

### LINKED ISSUE #1222: add bash completion for 'buildtest buildspec find --filter' and 'buildtest buildspec find --format'
Currently we don't have bash completion setup for `--filter` when this option is set we should auto-populate the filter fields instead of showing the `--*` options. 
```
(buildtest)  ~/Documents/github/buildtest/docs/ [devel] buildtest buildspec find --filter buildspec=$BUILDTEST_ROOT/tutorials/hello_world.yml
--buildspec          --format             --help               --no-header          --quiet              --tags               -e                   -p                   -t                                                         
--executors          --group-by-executor  --helpfilter         --pager              --rebuild            --terse              -h                   -q                   invalid                                                    
--filter             --group-by-tags      --helpformat         --paths              --root               -b                   -n                   -r                                                                            
```

This feature depends on #1221 to be completed.

### PR DESCRIPTION
**Pull request by prathmesh4321**

### CODE REVIEW

**Comment by prathmesh4321** [comment]
Bash completion for --format and --filter options
image

File: bash_completion.sh [hunk 1]
```diff
@@ -264,7 +264,18 @@ _buildtest ()
            local shortopts="-b -e -h -n -p -q -r -t"
            local subcmds="invalid"
            local allopts="${longopts} ${shortopts} ${subcmds}"
-           COMPREPLY=( $( compgen -W "${allopts}" -- $cur ) );;
+           COMPREPLY=( $( compgen -W "${allopts}" -- $cur ) )
+           case "${prev}" in --filter)
+             local opts="type executor tags buildspec"
```
Comment (shahzebsiddiqui) [inline bash_completion.sh]: please create a bash method that will call the following

```
(buildtest)  ~/Documents/github/buildtest/ [devel] buildtest buildspec find --filterfields
type
executor
tags
buildspec
```

File: bash_completion.sh [hunk 2]
```diff
@@ -264,7 +264,18 @@ _buildtest ()
...
+             COMPREPLY=( $( compgen -W "$opts" -- $cur ) )
+             return
+           esac
+           case "${prev}" in --format)
+             local opts="name type executor tags description buildspec"
```
Comment (shahzebsiddiqui) [inline bash_completion.sh]: same thing here call a method say `buildspec_format_fields` or whatever.

```
(buildtest)  ~/Documents/github/buildtest/ [devel] buildtest buildspec find --formatfields
name
type
executor
tags
description
buildspec
```

**Review by shahzebsiddiqui — CHANGES_REQUESTED** [review]
please dont hardcode these bash completion we should be using the builtin commands to get the output.

File: bash_completion.sh (hunk 2 above)
Comment (prathmesh4321) [reply bash_completion.sh]: Done ! @shahzebsiddiqui

**Review by shahzebsiddiqui — APPROVED** [review]
thanks @prathmesh4321 this looks good

**Pull request merged**